tests/wpt-jsdom/utils.js

```diff
@@ -0,0 +1,67 @@
+"use strict";
+const childProcess = require("child_process");
+const http = require("http");
+const https = require("https");
+const os = require("os");
+const { Canvas } = require("jsdom/lib/jsdom/utils");
+
+const hasCanvas = Boolean(Canvas);
+
+const nodeMajorVersion = process.versions.node.split(".")[0];
+
+exports.resolveReason = (reason) => {
+	if (["fail-slow", "timeout", "flaky"].includes(reason)) {
+		return "skip";
+	}
+
+	if (["fail-with-canvas", "needs-canvas"].includes(reason) && !hasCanvas) {
+		return "skip";
+	}
+
+	if (reason === "fail-node18" && nodeMajorVersion === "18") {
+		return "skip";
+	}
+
+	if (reason === "fail" || (reason === "fail-with-canvas" && hasCanvas)) {
+		return "expect-fail";
+	}
+
+	return "run";
+};
+
+exports.killSubprocess = (subprocess) => {
+	if (os.platform() === "win32") {
+		// subprocess.kill() doesn't seem to be able to kill descendant processes on Windows,
+		// at least with whatever's going on inside the web-platform-tests Python.
+		// Use this technique instead.
+		const { pid } = subprocess;
+		childProcess.spawnSync("taskkill", ["/F", "/T", "/PID", pid], {
+			detached: true,
+			windowsHide: true,
+		});
+	} else {
+		// SIGINT is necessary so that the Python script can clean up its subprocesses.
+		subprocess.kill("SIGINT");
+	}
+};
+
+// We need rejectUnauthorized support so we can't use built-in fetch(), sadly.
+exports.doHeadRequestWithNoCertChecking = (url) => {
+	const agent = url.startsWith("https")
+		? new https.Agent({ rejectUnauthorized: false })
```

Comment:
This is taken directly from the implementation in https://github.com/jsdom/jsdom/blob/2f8a730/test/. Let me know if feel strong that the tests need to prevent MITM attacks.

Reply:
Looks like an equivalent is already in use with `strictSSL` disabled, see https://github.com/eps1lon/dom-accessibility-api/pull/1032/files#diff-6a9bf1cc923f6f1ad64ad81fad823541d2729d2f9d9ea615e29c40a3a4efbaf0L85